### ISSUE TITLE
[GraphQL] Unable to share gift registry

### ISSUE THREAD

**Issue by magento-engcom-team** [magento/magento2#31639]
#### Pre-Requisites:
-  Create a gift registry

#### Steps to reproduce:
- Run the  shareGiftRegistry mutation for the logged in customer
```php
mutation{
  shareGiftRegistry
  (giftRegistryUid:"AKubNdI4QZWnejU6zl2zyoWQlGcTq1Aw" 
    sender:{message:"test" name:"John Doe"  } invitees:{email:"inviteee@wxample.com" name:"invitee1"}
    }
  )  { is_shared  }      } }
```

#### Expected Result
ShareGiftRegistryOutput returns true / false.

#### Actual Result
Exception is thrown :"We can't share the gift_registry"

P.S. This is happening in Luma as well.

**Comment by magento-engcom-team** [comment]
The issue was exported from the internal JIRA. The link to the original JIRA issue: https://jira.corp.magento.com/browse/MC-40402

**Comment by XxXgeoXxX** [comment]
@magento I am working on this

**Comment by eduard13** [comment]
Looks like the issue is somehow related to the GiftRegistry itself, as I'm not able to reproduce it locally.
![image](https://user-images.githubusercontent.com/15868188/104922403-5aa16500-59a3-11eb-9912-b48c7a0f8666.png)

**Comment by dthampy** [comment]
@eduard13 ,  This issue was originally reproduced on one of our cloud  staging instances. I later found out that this is caused by an environment variable for "Outgoing emails" being turned off for that instance. 
I checked the workflow locally and on the same instance with the above setting turned on. It is working as expected. 
Thank you for looking into it. 
@mauragcyrus , can you close this ticket, please? Thanks!

**Comment by magento-engcom-team** [comment]
Hi @magento-engcom-team, @shikhamis11, @XxXgeoXxX. 

Thank you for your report and collaboration!

The related internal Jira ticket `MC-40402` was closed as `Fixed`. 

The fix will be available with the upcoming `2.4.3` release.
